packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts

```diff
@@ -149,7 +149,25 @@ export class AwsCliCompatible {
  */
 async function getRegionFromIni(profile: string): Promise<string | undefined> {
   const sharedFiles = await loadSharedConfigFiles({ ignoreCache: true });
-  return sharedFiles?.configFile?.[profile]?.region || sharedFiles?.configFile?.default?.region;
+
+  // Priority:
+  //
+  // credentials come before config because aws-cli v1 behaves like that.
```

Comment:
Not entirely true that AWS CLI v1 behaves this way, but we're ~consistent with how users will *probably* use it 😉